### PR TITLE
feat: add activity message for SLA

### PR DESCRIPTION
**Pull request by vishnu-narayanan**
# Pull Request Template

## Description

Fixes https://linear.app/chatwoot/issue/CW-3121/sla-activity-message

Changelog
----
- Generate activity messages if SLA is added/removed from a conversation
- Refactor Label activity messages methods to a separate concern to bring down module length
- Refactor Team activity messages methods to a separate concern to bring down module length
- 

image

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Tested locally by creating a new conversation. The automation rule added SLA policy to the conversation, generating an SLA policy activity message.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by sojan-official — COMMENTED** [review]
Code LGTM . lets add some specs to cover this case.

**Comment by vishnu-narayanan** [comment]
@sojan-official Done. Could you please take another look?

File: app/models/concerns/activity_message_handler.rb [hunk 1]
```diff
@@ -119,4 +138,15 @@ def activity_message_ownner(user_name)
     user_name = 'Automation System' if !user_name && Current.executed_by.present?
     user_name
   end
+
+  def determine_sla_change_type
```
Comment (scmmishra) [inline app/models/concerns/activity_message_handler.rb:142]: We can move this method inside `SlaActivityMessageHandler` and reduce the lines of code for this file, that way we won't need any `rubocop:disable`

**Review by scmmishra — CHANGES_REQUESTED** [review]
Code looks good, lets try to enable the `Metrics/ModuleLength` back

File: app/models/concerns/activity_message_handler.rb (hunk 1 above)
Comment (vishnu-narayanan) [reply app/models/concerns/activity_message_handler.rb]: Let me check

**Review by scmmishra — APPROVED** [review]

**Pull request merged**